src/d/actor/d_a_warpfout.cpp

```diff
@@ -38,91 +64,187 @@ void daWarpfout_c::eventOrder() {
+        return TRUE;
+    } else {
+        return FALSE;
+    }
 }
 
 /* 000003EC-00000468       .text initWarp3__12daWarpfout_cFi */
-void daWarpfout_c::initWarp3(int) {
-    /* Nonmatching */
+BOOL daWarpfout_c::initWarp3(int) {
+    cXyz effectPos = dComIfGp_getLinkPlayer()->current.pos;
+    effectPos.y += 80.0f;
+
+    set_effect_wind01(effectPos, 0x71c);
+    mTimer = 5;
 }
 
 /* 00000468-00000494       .text actWarp3__12daWarpfout_cFi */
-void daWarpfout_c::actWarp3(int) {
-    /* Nonmatching */
+BOOL daWarpfout_c::actWarp3(int) {
+    if (!cLib_calcTimer(&mTimer)) {
+        return TRUE;
+    } else {
+        return FALSE;
+    }
 }
 
 /* 00000494-0000051C       .text initWarp4__12daWarpfout_cFi */
-void daWarpfout_c::initWarp4(int) {
-    /* Nonmatching */
+BOOL daWarpfout_c::initWarp4(int) {
+    fopAc_ac_c* link = dComIfGp_getLinkPlayer();
+
+    for (int i = 0; i < 6; i++) {
+        dComIfGp_particle_set(0x830e, &link->current.pos);
+    }
 }
 
 /* 0000051C-00000524       .text actWarp4__12daWarpfout_cFi */
-void daWarpfout_c::actWarp4(int) {
-    /* Nonmatching */
+BOOL daWarpfout_c::actWarp4(int) {
+    return TRUE;
 }
 
 /* 00000524-00000528       .text initEnd__12daWarpfout_cFi */
-void daWarpfout_c::initEnd(int) {
-    /* Nonmatching */
-}
+BOOL daWarpfout_c::initEnd(int) {}
 
 /* 00000528-0000054C       .text actEnd__12daWarpfout_cFi */
-void daWarpfout_c::actEnd(int) {
-    /* Nonmatching */
+BOOL daWarpfout_c::actEnd(int) {
+    fopAcM_delete(this);
+    return TRUE;
 }
 
 /* 0000054C-000005F4       .text set_effect_wind01__12daWarpfout_cF4cXyzs */
-void daWarpfout_c::set_effect_wind01(cXyz, short) {
-    /* Nonmatching */
+void daWarpfout_c::set_effect_wind01(cXyz effectPos, s16 unk) {
+    csXyz effectAngle = get_effect_angle();
+    effectAngle.z += unk;
+
+    // set__13dPa_control_cFUcUsPC4cXyzPC5csXyzPC4cXyzUcP18dPa_levelEcallBackScPC8_GXColorPC8_GXColorPC4cXyz
+    dComIfGp_particle_set(
+        /* particleId = */ 0x830f,
+        /* pos        = */ &effectPos,
+        /* angle      = */ &effectAngle);
 }
 
 /* 000005F4-00000670       .text get_effect_angle__12daWarpfout_cFv */
-void daWarpfout_c::get_effect_angle() {
-    /* Nonmatching */
+csXyz daWarpfout_c::get_effect_angle() {
+    csXyz angle = current.angle;
+
+    angle.x = dCam_getAngleX(dComIfGp_getCamera(0));
+    angle.y = dCam_getAngleY(dComIfGp_getCamera(0)) + 0x8000;
+    angle.z = 0;
+
+    return angle;
 }
 
 /* 00000670-00000690       .text daWarpfout_Create__FPv */
-static s32 daWarpfout_Create(void*) {
-    /* Nonmatching */
+static s32 daWarpfout_Create(void* i) {
+    daWarpfout_c* i_this = static_cast<daWarpfout_c*>(i);
+
+    return i_this->_create();
 }
 
 /* 00000690-000006B4       .text daWarpfout_Delete__FPv */
-static BOOL daWarpfout_Delete(void*) {
-    /* Nonmatching */
+static BOOL daWarpfout_Delete(void* i) {
+    daWarpfout_c* i_this = static_cast<daWarpfout_c*>(i);
+
+    return i_this->_delete() & 0xff;
```

Comment:
Try declaring `_delete` and `_execute` as `bool` rather than `BOOL`.

Reply:
the `& 0xff` can be removed now that `_delete` returns `bool`